infra/feast-operator/internal/controller/services/util.go

```diff
@@ -0,0 +1,94 @@
+}
+
+func ApplyDefaultsToStatus(cr *feastdevv1alpha1.FeatureStore) {
+	cr.Status.FeastVersion = feastversion.FeastVersion
+	applied := cr.Spec.DeepCopy()
+	if applied.Services == nil {
+		applied.Services = &feastdevv1alpha1.FeatureStoreServices{}
+	}
+
+	// default to registry service deployment
+	if applied.Services.Registry == nil {
+		applied.Services.Registry = &feastdevv1alpha1.Registry{
+			Local: &feastdevv1alpha1.LocalRegistryConfig{
+				Persistence: &feastdevv1alpha1.RegistryPersistence{
+					FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
+						Path: DefaultRegistryPath,
+					},
+				},
+			},
+		}
+	}
+	// if remote registry not set, proceed w/ local registry defaults
+	if applied.Services.Registry.Remote == nil {
+		// if local registry not set, apply an empty pointer struct
+		if applied.Services.Registry.Local == nil {
+			applied.Services.Registry.Local = &feastdevv1alpha1.LocalRegistryConfig{
+				Persistence: &feastdevv1alpha1.RegistryPersistence{
+					FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
+						Path: DefaultRegistryPath,
+					},
+				},
+			}
+		}
+		if applied.Services.Registry.Local.Persistence == nil {
+			applied.Services.Registry.Local.Persistence = &feastdevv1alpha1.RegistryPersistence{
+				FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
+					Path: DefaultRegistryPath,
+				},
+			}
+		} else if applied.Services.Registry.Local.Persistence.FilePersistence == nil {
+			applied.Services.Registry.Local.Persistence.FilePersistence = &feastdevv1alpha1.RegistryFilePersistence{
+				Path: DefaultRegistryPath,
+			}
+		}
+		setServiceDefaultConfigs(&applied.Services.Registry.Local.ServiceConfigs.DefaultConfigs)
+	}
+	if applied.Services.OfflineStore != nil {
+		setServiceDefaultConfigs(&applied.Services.OfflineStore.ServiceConfigs.DefaultConfigs)
+		if applied.Services.OfflineStore.Persistence == nil {
+			applied.Services.OfflineStore.Persistence = &feastdevv1alpha1.OfflineStorePersistence{
+				FilePersistence: &feastdevv1alpha1.OfflineStoreFilePersistence{
+					Type: string(OfflineDaskConfigType),
+				},
+			}
+		} else if applied.Services.OfflineStore.Persistence.FilePersistence == nil {
+			applied.Services.OfflineStore.Persistence.FilePersistence = &feastdevv1alpha1.OfflineStoreFilePersistence{
+				Type: string(OfflineDaskConfigType),
+			}
+		}
+	}
+	if applied.Services.OnlineStore != nil {
+		setServiceDefaultConfigs(&applied.Services.OnlineStore.ServiceConfigs.DefaultConfigs)
+		if applied.Services.OnlineStore.Persistence == nil {
+			applied.Services.OnlineStore.Persistence = &feastdevv1alpha1.OnlineStorePersistence{
+				FilePersistence: &feastdevv1alpha1.OnlineStoreFilePersistence{
+					Path: DefaultOnlinePath,
+				},
+			}
+		} else if applied.Services.OnlineStore.Persistence.FilePersistence == nil {
+			applied.Services.OnlineStore.Persistence.FilePersistence = &feastdevv1alpha1.OnlineStoreFilePersistence{
+				Path: DefaultOnlinePath,
+			}
+		}
+	}
```

Comment:
same

Reply:
would be good to have unit tests ensuring that proper defaults are being set in status when varying pointers around service persistence are set in the spec.

Reply:
that's a addition, and necessary, but its different from ensuring the status defaults are properly being reconciled in the CR.

Reply:
I see what you mean, I'll add it in the next PR for PVC so there will be more cases to cover